src/fast_align.cc

```diff
@@ -317,6 +319,104 @@ void InitialPass(const unsigned kNULL, const bool use_null, TTable* s2t,
   cerr << "expected target length = source length * " << mean_srclen_multiplier << endl;
 }
 
+double ForceAlign(const string& line, int lc, double prob_align_not_null, bool use_null, string& ret, const unsigned kNULL, const TTable& s2t)
```

Comment:
This is basically lifted straight out of main(). Only changes I made were:

(1) Pre-reserve vectors of (2x) sufficient size so we don't end up doing reallocations.
(2) Use itoa + string buffers instead of stream IO when outputting alignments (using stream buffers blocks on facet retrieval).